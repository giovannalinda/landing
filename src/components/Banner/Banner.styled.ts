import styled from 'styled-components'
import { Assets } from '~/assets'
import { theme } from '~/styles'
import { Container as Button } from '../Button/Button.styled'

export const Container = styled.section`
  background-image: url(${Assets.Shapes});
  height: 31.2rem;
  max-width: ${theme.grid.lg};
  width: 100%;
  margin: 16.9rem auto 0;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`

export const TextGroup = styled.div`
  max-width: 60rem;
  padding-bottom: ${theme.space[6]};

  p {
    text-align: center;
    font-size: ${theme.font.sizes.subtitle};
    font-weight: ${theme.font.weigths.regular};
    line-height: ${theme.font.lineHeights.subtitle};

    span {
      color: ${theme.colors.secondaryPink};
    }
  }
`

export const HireMeButton = styled(Button)`
  display: flex;
  align-items: center;

  svg {
    margin-left: ${theme.space[2]};
  }
`
