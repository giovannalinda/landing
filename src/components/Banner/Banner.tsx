import { CgArrowLongRight } from 'react-icons/cg'
import { useTranslation } from '~/lib/useTranslation'

import * as S from './Banner.styled'

export function Banner() {
  const { t } = useTranslation()

  return (
    <S.Container>
      <S.TextGroup>
        <p>
          {`${t("Got a project in mind? Let's make something awesome")} `}
          <span>{t('together')}</span>.
        </p>
      </S.TextGroup>
      <S.HireMeButton variant='purple' size='md' rounded={false}>
        {t('Hire me')} <CgArrowLongRight size={20} />
      </S.HireMeButton>
    </S.Container>
  )
}
