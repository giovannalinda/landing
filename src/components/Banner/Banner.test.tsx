import { render } from '~/__helpers__/app-tests'
import { Banner } from './Banner'

describe('Banner', () => {
  it('should match snapshot', () => {
    const { container } = render(<Banner />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
