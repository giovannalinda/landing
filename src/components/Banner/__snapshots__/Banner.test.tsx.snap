// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should match snapshot 1`] = `
<section
  class="sc-gsnTZi hmNYdX"
>
  <div
    class="sc-dkzDqf iHatTL"
  >
    <p>
      Got a project in mind? Let's make something awesome 
      <span>
        together
      </span>
      .
    </p>
  </div>
  <button
    class="sc-bczRLJ sc-hKMtZM fhxDpE bknSvH"
  >
    Hire me
     
    <svg
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M23.0677 11.9929L18.818 7.75739L17.4061 9.17398L19.2415 11.0032L0.932469 11.0012L0.932251 13.0012L19.2369 13.0032L17.4155 14.8308L18.8321 16.2426L23.0677 11.9929Z"
        fill="currentColor"
      />
    </svg>
  </button>
</section>
`;
